scapy/layers/l2.py

```diff
@@ -807,33 +809,45 @@ def arp_mitm(
         $ sudo scapy
         >>> arp_mitm("192.168.122.156", "192.168.122.17")
 
+    Alternative usages:
+        >>> arp_mitm("10.0.0.1", "10.1.1.0/21", iface="eth1")
+        >>> arp_mitm("10.0.0.1", "10.1.1.2",
+        ...          target_mac="aa:aa:aa:aa:aa:aa",
+        ...          mac2="00:1e:eb:bf:c1:ab")
+
     Remember to change the sysctl settings back..
     """
     if not iface:
         iface = conf.route.route(ip1)[0]
     if not target_mac:
         target_mac = get_if_hwaddr(iface)
     if mac1 is None:
-        mac1 = getmacbyip(ip1)
-        if not mac1:
-            print("Can't resolve mac for %s" % ip1)
-            return
+        if "/" in ip1:  # subnet
+            mac1 = "ff:ff:ff:ff:ff:ff"
+        else:
+            mac1 = getmacbyip(ip1)
+            if not mac1:
+                print("Can't resolve mac for %s" % ip1)
+                return
     if mac2 is None:
-        mac2 = getmacbyip(ip2)
-        if not mac2:
-            print("Can't resolve mac for %s" % ip2)
-            return
+        if "/" in ip2:  # subnet
+            mac2 = "ff:ff:ff:ff:ff:ff"
+        else:
+            mac2 = getmacbyip(ip2)
+            if not mac2:
+                print("Can't resolve mac for %s" % ip2)
+                return
     print("MITM on %s: %s <--> %s <--> %s" % (iface, mac1, target_mac, mac2))
     # We loop who-has requests
     srploop(
-        [
+        list(itertools.chain(
             Ether(dst=mac1, src=target_mac) /
             ARP(op="who-has", psrc=ip2, pdst=ip1,
                 hwsrc=target_mac, hwdst="ff:ff:ff:ff:ff:ff"),
```

Comment:
This is not in the PR but still: I **think** it should be `"00:00:00:00:00:00"` here and below.

Reply:
Indeed !

Reply:
Do you want to use the opportunity of this PR to change it (and in the other packet below too)?

Reply:
Yes ! I had forgotten to commit some changes :/ Should now be included